packages/services/pkg/mode/storecore/encoding.go

```diff
@@ -0,0 +1,631 @@
+		value := DecodeStaticField(fieldType, encoding, bytesOffset)
+		bytesOffset += getStaticByteLength(fieldType)
+
+		// Save a mapping of FIELD TYPE (string) -> (FIELD VALUE (interface{}), FIELD TYPE (SchemaType))
+		data.Set(fieldType.String(), &DataSchemaTypePair{
+			Data:       value,
+			SchemaType: fieldType,
+		})
+	}
+
+	// Decode dynamic fields.
+	if len(schemaTypePair.Dynamic) > 0 {
+		dynamicDataSlice := encoding[schemaTypePair.StaticDataLength : schemaTypePair.StaticDataLength+32]
+		bytesOffset += 32
+
+		for i, fieldType := range schemaTypePair.Dynamic {
+			offset := 4 + i*2
+			dataLength := new(big.Int).SetBytes(dynamicDataSlice[offset : offset+2]).Uint64()
+			value := DecodeDynamicField(fieldType, encoding[bytesOffset:bytesOffset+dataLength])
+			bytesOffset += dataLength
+
+			// Save a mapping of FIELD TYPE (string) -> (FIELD VALUE (interface{}), FIELD TYPE (SchemaType))
+			data.Set(fieldType.String(), &DataSchemaTypePair{
+				Data:       value,
+				SchemaType: fieldType,
+			})
+
+		}
+	}
+
+	return data
+}
+
+// DecodeDynamicField decodes the dynamic field of the specified schema type from the given byte slice encoding.
+//
+// Parameters:
+// - schemaType (SchemaType): The schema type of the dynamic field to be decoded.
+// - encodingSlice ([]byte): The byte slice encoding of the dynamic field.
+//
+// Returns:
+// - (string): The decoded value of the dynamic field as a string.
+func DecodeDynamicField(schemaType SchemaType, encodingSlice []byte) string {
+	switch schemaType {
+	case BYTES:
+		return hexutil.Encode(encodingSlice)
+	case STRING:
+		return string(encodingSlice)
+	default:
+		logger.GetLogger().Fatal("Unknown dynamic field type", zap.String("type", schemaType.String()))
+		return ""
+	}
+}
+
+// DecodeStaticField decodes a static field of a given schema type from a byte slice.
+//
+// Parameters:
+// - schemaType (SchemaType): The schema type of the field to decode.
+// - encoding ([]byte): The byte slice containing the encoded field.
+// - bytesOffset (uint64): The offset in bytes within the encoding byte slice where the field to be decoded starts.
+//
+// Returns:
+// - (string): The decoded field as a string.
+func DecodeStaticField(schemaType SchemaType, encoding []byte, bytesOffset uint64) string {
+	switch schemaType {
+	case UINT8:
+		return fmt.Sprint(new(big.Int).SetBytes(encoding[bytesOffset : bytesOffset+1]).Uint64())
+	case UINT16:
+		return fmt.Sprint(new(big.Int).SetBytes(encoding[bytesOffset : bytesOffset+2]).Uint64())
+	case UINT32:
+		return fmt.Sprint(new(big.Int).SetBytes(encoding[bytesOffset : bytesOffset+4]).Uint64())
+	case UINT64:
+		return fmt.Sprint(new(big.Int).SetBytes(encoding[bytesOffset : bytesOffset+8]).Uint64())
+	case UINT128:
+		return fmt.Sprint(new(big.Int).SetBytes(encoding[bytesOffset : bytesOffset+16]).Uint64())
+	case UINT256:
+		return fmt.Sprint(new(big.Int).SetBytes(encoding[bytesOffset : bytesOffset+32]).Uint64())
```

Comment:
would the conversion to uint64 here fail if the number encoded is bigger than 2^64? (which can be the case with uint128 and uin256)